optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/roster/RosterRestService.java

```diff
@@ -20,51 +22,75 @@
 
 import io.swagger.annotations.Api;
 
-@Api(tags = { "Roster" })
+@Api(tags = {"Roster"})
 @Path("/tenant/{tenantId}/roster")
 @Produces(MediaType.APPLICATION_JSON)
 @Consumes(MediaType.APPLICATION_JSON)
 @GenRestBuilder
 public interface RosterRestService {
 
+    // ************************************************************************
+    // SpotRosterView
+    // ************************************************************************
+
+    @ApiOperation("Get the current spot roster view")
     @GET
     @Path("/spotRosterView/current")
-    SpotRosterView getCurrentSpotRosterView(@PathParam("tenantId") Integer tenantId);
+    SpotRosterView getCurrentSpotRosterView(@ApiParam(required = true) @PathParam("tenantId") Integer tenantId);
 
+    @ApiOperation("Get a spot roster view between 2 dates")
     @GET
     @Path("/spotRosterView")
-    SpotRosterView getSpotRosterView(@PathParam("tenantId") Integer tenantId,
-            @QueryParam("startDate") String startDateString, @QueryParam("endDate") String endDateString);
+    SpotRosterView getSpotRosterView(@ApiParam(required = true) @PathParam("tenantId") Integer tenantId,
+            @ApiParam(value = "inclusive", required = true) @QueryParam("startDate") String startDateString,
+            @ApiParam(value = "exclusive", required = true) @QueryParam("endDate") String endDateString);
 
     //TODO: find out if there a way to pass lists in GET requests
+    @ApiOperation("Get a spot roster view between 2 dates for a subset of the spots")
     @POST
-    @Path("/spotRosterView/for")
-    SpotRosterView getSpotRosterViewFor(@PathParam("tenantId") Integer tenantId,
-            @QueryParam("startDate") String startDateString, @QueryParam("endDate") String endDateString,
-            List<Spot> spots);
+    @Path("/spotRosterView/for") // TODO rename to forSpots and send only the spot id's
```

Comment:
I've changed the comment to be clearer and focus on the problem (instead of the solution):

 // TODO naming "for" is too abstract: we might add a sibling rest method that filters on another type than spots too